### ISSUE TITLE
Display objects from component dependency differently

### ISSUE THREAD

**Issue by tir38** [arunkumar9t2/scabbard#22]
Part of our overall object graph looks like this:

![io orionlabs android feature media di MediaComponent](https://user-images.githubusercontent.com/1980553/71684750-5092d100-2d64-11ea-861a-ffe7c1a4575b.png)

Some of these objects in the graph are coming from a component dependency:

```
@Component(
        modules = {
                MediaModule.class
        },
        dependencies = {
                SomeOtherComponent.class
        }
)
public interface MediaComponent {...}
```

You can probably infer which ones are coming from SomeOtherComponent based on package name. it would be nice if these were either color coded differently than those supplied by `MediaModule`, or maybe even put them in another box *inside* "object graph".

**Comment by arunkumar9t2** [comment]
Agreed, although package name might not be a universal solution for all projects. I'll check if there is something in the Dagger SPI to differentiate dependencies.

**Comment by tir38** [comment]
I just meant in my case the package name would show *you* what I'm talking about. I don't think relying on package name to infer component is the right *programmatic* solution. =)

**Comment by arunkumar9t2** [comment]
This was blocked by https://github.com/google/dagger/issues/3483. Now that is implemented in Dagger, I will try to display component dependencies differently.
